src/modules/CRUD/components/Details/Read/AttachmentView/AttachmentList/AttachmentList.js

```diff
@@ -1,36 +1,80 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
-import classNames from 'classnames';
+import classnames from 'classnames';
+import Loading from '../../../../../../../components/Loading';
 
 import Attachment from '../Attachment';
 import Picture from '../Picture';
+import Figcaption from '../Figcaption';
+import DeleteAttachment from '../DeleteAttachment';
 
 import './styles.scss';
 
 const AttachmentList = ({
-  name,
   attachments,
   className,
+  featureEndpoint,
+  fetchFeature,
+  history,
+  location,
+  match: { params: { id: paramId } },
+  name,
+  staticContext,
   ...props
 }) => {
+  const [isDeleting, setDeleting] = useState(null);
+
+  const updateData = useCallback(async () => {
+    fetchFeature(featureEndpoint, paramId);
+  }, [featureEndpoint, fetchFeature, paramId]);
+
   const Component = name === 'attachments' ? Attachment : Picture;
   return (
-    <ul className={classNames('attachment-list', className)} {...props}>
-      {attachments.map(({ id, file, legend: label, thumbnail }) => (
-        <li className="attachment-list__item attachment-item" key={id}>
-          <figure className="attachment-item__figure">
-            <Component
-              className="attachment-item__file"
-              file={file}
+    <ul className={classnames('attachment-list', className)} {...props}>
+      {attachments.map(({
+        action_url: endpoint,
+        category,
+        id,
+        file,
+        legend: label,
+        thumbnail,
+      }) => {
+        const isLoading = id === isDeleting;
+        return (
+          <li
+            className={classnames({
+              'attachment-list__item': true,
+              'attachment-item': true,
+              'attachment-item--loading': isLoading,
+            })}
+            key={id}
+          >
+            {isLoading && <Loading className="attachment-item__Loading" spinner />}
+            <figure className="attachment-item__figure">
+              <Component
+                className="attachment-item__file"
+                file={file}
+                label={label}
+                thumbnail={thumbnail}
+              />
+              <Figcaption
+                category={category}
+                endpoint={endpoint}
+                label={label}
+                updateData={updateData}
+              />
+            </figure>
+            <DeleteAttachment
+              endpoint={endpoint}
+              id={id}
               label={label}
-              thumbnail={thumbnail}
+              loading={isLoading}
+              setDeleting={setDeleting}
+              updateData={updateData}
             />
-            <figcaption className="attachment-item__figcaption">
-              <span className="attachment-item__figcaption-text">{label}</span>
-            </figcaption>
-          </figure>
-        </li>
-      ))}
+          </li>
+        );
```

Comment:
Couldn't this be in another (sub)component ? (like `AttachmentItem`)

Reply:
I just made the split